jina/hubble/helper.py

```diff
@@ -287,10 +282,18 @@ def wrapper(*args, **kwargs):
             call_hash = f'{func.__name__}({", ".join(map(str, args))})'
 
             pickle_protocol = 4
-            import filelock
+            file_lock = nullcontext()
+            with ImportExtensions(
+                required=False,
+                help_text=f'FileLock is needed to guarantee non-concurrent access to the'
+                f'cache_file {cache_file}',
+            ):
+                import filelock
+
+                file_lock = filelock.FileLock(f'{cache_file}.lock', timeout=-1)
 
             cache_db = None
-            with filelock.FileLock(f'{cache_file}.lock', timeout=-1):
+            with file_lock:
```

Comment:
Should we even allow a nullcontext? Maybe make required=True?

Reply:
required True means an exception will be raised and failed.

Reply:
Yes, that was my point. Shouldn't make this required? What happens if you don't have the actual package and just have a nullcontext?

Reply:
in a very rare case where Flows are run in multiprocess or multithreaded way